xbmc/ServiceManager.cpp

```diff
@@ -0,0 +1,58 @@
+/*
+ *      Copyright (C) 2005-2016 Team XBMC
+ *      http://xbmc.org
+ *
+ *  This Program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2, or (at your option)
+ *  any later version.
+ *
+ *  This Program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with XBMC; see the file COPYING.  If not, see
+ *  <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#include "ServiceManager.h"
+#include "utils/log.h"
+
+CServiceMgr::~CServiceMgr()
+{
+  m_binaryAddonCache.reset();
+  m_addonMgr.reset();
+}
+
+bool CServiceMgr::Init()
+{
+  m_addonMgr.reset(new ADDON::CAddonMgr());
+  if (!m_addonMgr->Init())
+  {
+    CLog::Log(LOGFATAL, "CServiceMgr::Init: Unable to start CAddonMgr");
+    return false;
+  }
+  m_binaryAddonCache.reset( new ADDON::CBinaryAddonCache());
+  m_binaryAddonCache->Init();
+
+  return true;
+}
+
+void CServiceMgr::Deinit()
+{
+  m_binaryAddonCache.reset();
+  m_addonMgr.reset();
+}
+
+ADDON::CAddonMgr *CServiceMgr::GetAddonMgr()
```

Comment:
I was thinking about placing an assert here. if this is nullptr something went wrong in a way that app won't run